cmake/package_nsis.cmake

```diff
@@ -44,47 +44,88 @@ else()
   set(PLATFORM "64")
   set(ARCH "x64")
 endif()
+
 if (NOT VC_REDIST_EXE)
-  set(VC_REDIST_EXE "vcredist_${ARCH}.exe")
+	set(VC_REDIST_EXE "vcredist_${ARCH}.exe")
 endif()
 
-## Find redistributable to be installed by NSIS
+# Find redistributable to be installed by NSIS
 if (NOT VC_REDIST_PATH)
-  string(REGEX REPLACE ".*Visual Studio ([1-9][1-9]).*" "\\1" OPENMS_MSVC_VERSION_STRING "${CMAKE_GENERATOR}")
-  if("${OPENMS_MSVC_VERSION_STRING}" GREATER "15")
-    if (DEFINED ENV{VCINSTALLDIR})
-      ## according to https://docs.microsoft.com/de-de/cpp/windows/redistributing-visual-cpp-files?view=msvc-160
-      get_filename_component(VC_ROOT_PATH "$ENV{VCINSTALLDIR}Redist/MSVC/v142" ABSOLUTE)
-      file(GLOB_RECURSE VC_REDIST_ABS_PATH "${VC_ROOT_PATH}/${VC_REDIST_EXE}")
-      message(STATUS "VS Redist locations (1st try): ${VC_REDIST_ABS_PATH}")
-    endif()
-    if (NOT VC_REDIST_ABS_PATH AND DEFINED ENV{VCToolsRedistDir}) ## if still not found, try to use older methods
-      ## An alternative solution to find redists
-      #execute_process(COMMAND "$ENV{PROGRAMFILES}/Microsoft Visual Studio/Installer/vswhere" -latest -version "${OPENMS_MSVC_VERSION_STRING}" -property installationPath
-      #                OUTPUT_VARIABLE VC_ROOT_PATH
-      #				  ERROR_VARIABLE VSWHERE_ERROR
-      #				  RESULT_VARIABLE VSWHERE_RESULT)
-      #if ("${VSWHERE_RESULT}" NOTEQUAL "0")
-      #  message(FATAL_ERROR "Executing vswhere to find vsredist executable for win packaging failed. Either specify VC_REDIST_PATH or make sure vswhere works.")
-      #endif()
-
-      ## We have to glob recurse in the parent folder because there is a version number in the end.
-      ## Unfortunately in my case the default version (latest) does not include the redist?!
-      ## TODO Not sure if this environment variable always exists. In the VS command line it should! Fallback vswhere or VCINSTALLDIR/Redist/MSVC?
-      get_filename_component(VC_ROOT_PATH "$ENV{VCToolsRedistDir}.." ABSOLUTE)
-      file(GLOB_RECURSE VC_REDIST_ABS_PATH "${VC_ROOT_PATH}/${VC_REDIST_EXE}")
-      message(STATUS "VS Redist locations (2nd try): ${VC_REDIST_ABS_PATH}")
-    endif()
-    if (VC_REDIST_ABS_PATH)
-      ## arbitrarily pick first of the found redists
-      list(GET VC_REDIST_ABS_PATH 0 VC_REDIST_ABS_PATH)
-      get_filename_component(VC_REDIST_PATH "${VC_REDIST_ABS_PATH}" DIRECTORY)
-      message(STATUS "   ... picked first directory: ${VC_REDIST_PATH}")
-    endif()
-  endif()
-  if (NOT VC_REDIST_PATH) ## if still not found
-    message(FATAL_ERROR "Variable VC_REDIST_PATH missing. Are you on a proper VS 2019+ command line?")
-  endif()
+	if (MSVC_TOOLSET_VERSION EQUAL 141)
+		set(VS_VERSION 15)
+	elseif(MSVC_TOOLSET_VERSION EQUAL 142)
+		set(VS_VERSION 16)
+	elseif(MSVC_TOOLSET_VERSION EQUAL 143)
```

Comment:
this list seems to be subject to change:
"Compiler versions newer than those known to CMake will be reported as the latest known toolset version."
Where should we document that this potentially needs an upgrade?

Reply:
https://gitlab.kitware.com/cmake/cmake/-/issues/24230